resources/asciidoctor/spec/copy_images_spec.rb

```diff
@@ -0,0 +1,225 @@
+          ::File.join(tmp, 'tmp_example1.png')
+      )
+
+      copied = []
+      attributes = copy_attributes copied
+      attributes['resources'] = tmp
+      input = <<~ASCIIDOC
+        == Example
+        image::tmp_example1.png[]
+      ASCIIDOC
+      expect { convert input, attributes }.to raise_error { |error|
+        expect(error).to be_a(ConvertError)
+        expect(error.message).to match(/INFO: <stdin>: line 2: copying tmp_example1.png/)
+        # NOCOMMIT full paths in logs too, I think
+      }
+      expect(copied).to eq([
+          ["tmp_example1.png", "#{tmp}/tmp_example1.png"]
+      ])
+    }
+  end
+
+  it "can find files using a multi valued resources attribute" do
+    Dir.mktmpdir {|tmp|
+      FileUtils.cp(
+          ::File.join(spec_dir, 'resources', 'copy_images', 'example1.png'),
+          ::File.join(tmp, 'tmp_example1.png')
+      )
+
+      copied = []
+      attributes = copy_attributes copied
+      attributes['resources'] = "dummy1,#{tmp},/dummy2"
+      input = <<~ASCIIDOC
+        == Example
+        image::tmp_example1.png[]
+      ASCIIDOC
+      expect { convert input, attributes }.to raise_error { |error|
+        expect(error).to be_a(ConvertError)
+        expect(error.message).to match(/INFO: <stdin>: line 2: copying tmp_example1.png/)
+      }
+      expect(copied).to eq([
+          ["tmp_example1.png", "#{tmp}/tmp_example1.png"]
+      ])
+    }
+  end
+
+  it "has a nice error message when it can't find a file with single valued resources attribute" do
+    Dir.mktmpdir {|tmp|
+      copied = []
+      attributes = copy_attributes copied
+      attributes['resources'] = tmp
+      input = <<~ASCIIDOC
+        == Example
+        image::not_found.png[]
+      ASCIIDOC
+      expect { convert input, attributes }.to raise_error { |error|
+        expect(error).to be_a(ConvertError)
+        expect(error.message).to match(/
+            WARN:\ <stdin>:\ line\ 2:\ can't\ read\ image\ at\ any\ of\ \[
+              "#{spec_dir}\/not_found.png",\s
+              "#{tmp}\/not_found.png"
+              .+
+            \]/x)
+        expect(error.message).not_to match(/INFO: <stdin>/)
+      }
+      expect(copied).to eq([])
+    }
+  end
+
+  it "has a nice error message when it can't find a file with multi valued resources attribute" do
+    Dir.mktmpdir {|tmp|
+      copied = []
+      attributes = copy_attributes copied
+      attributes['resources'] = "#{tmp},/dummy2"
+      input = <<~ASCIIDOC
+        == Example
+        image::not_found.png[]
+      ASCIIDOC
+      expect { convert input, attributes }.to raise_error { |error|
+        expect(error).to be_a(ConvertError)
+        expect(error.message).to match(/
+            WARN:\ <stdin>:\ line\ 2:\ can't\ read\ image\ at\ any\ of\ \[
+              "#{spec_dir}\/not_found.png",\s
+              "#{tmp}\/not_found.png",\s
+              "\/dummy2\/not_found.png"
+              .+
```

Comment:
We list out all of the directories that we try and I don't think that list is worth maintaining here. I *do* think it is helpful to see it when it fails. Though I don't like how long the list gets.....